### PR TITLE
Landing pads

### PR DESCRIPTION
**Pull request by sboigelot**
A small landing pad for trading.

The landing pad currently allow to trade steel for steel, with a small profit, at the cost of work time.

The behaviour is a temporary behaviour to be expanded with futher trade system.

image

### CODE REVIEW

**Comment by sboigelot** [comment]
If someone know how to debug the lua code or have more info about the exception, I'm searching for info :D

I'll investigate more tonigth

File: Assets/StreamingAssets/LUA/Furniture.lua [hunk 1]
```diff
@@ -307,4 +307,66 @@ function CloningPod_JobComplete(j)
...
+			
+			furniture.AddJob(j)
+		end
+	else
+		furniture.ChangeParameter("smelttime", deltaTime)
+		
+		if(furniture.GetParameter("smelttime") >= furniture.GetParameter("smelttime_required")) then
+			furniture.SetParameter("smelttime", 0)
+			
+		 --[[	outputSpot = World.current.GetTileAt(spawnSpot.X+2, spawnSpot.y)
```
Comment (OzieGamma) [inline Assets/StreamingAssets/LUA/Furniture.lua]: Is this commented code?

Can't we delete it?

Comment (relinu) [inline Assets/StreamingAssets/LUA/Furniture.lua]: and shouldnt it be "tradetime" and "tradetime_required"?

Comment (sboigelot) [inline Assets/StreamingAssets/LUA/Furniture.lua]: It's WIP code, I'm trying to figure out where the nil exception comes from by commenting code.

Comment (sboigelot) [inline Assets/StreamingAssets/LUA/Furniture.lua]: @relinu yes that must be my nil exception, I was watching the rest! That's why I shared it, a second eye is priceless! Thanks a lot, I'll try to fix it now

Comment (sboigelot) [inline Assets/StreamingAssets/LUA/Furniture.lua]: no :( I still get the issue, even while commenting more lua code

ScriptRuntimeException: attempt to index a nil value
MoonSharp.Interpreter.Execution.VM.Processor.ExecIndex (MoonSharp.Interpreter.Execution.VM.Instruction i, Int32 instructionPtr)
MoonSharp.Interpreter.Execution.VM.Processor.Processing_Loop (Int32 instructionPtr)


I'll add a check in the parameter get/change in c# to throw a proper log in that case, but I may have to pause a few hours first

Comment (relinu) [inline Assets/StreamingAssets/LUA/Furniture.lua]: @sboigelot can you update the code here to the "newest" version?

Comment (sboigelot) [inline Assets/StreamingAssets/LUA/Furniture.lua]: the problem seems to be on this line:


function LandingPad_Temp_JobComplete(j)
	**spawnSpot = j.tile.furniture.GetSpawnSpotTile()**

no idea why for now

Comment (OzieGamma) [inline Assets/StreamingAssets/LUA/Furniture.lua]: @sboigelot I suggest avoiding in progress PRs. People will test out your code when you're not finished yet, losing time.

Great work though!

Comment (relinu) [inline Assets/StreamingAssets/LUA/Furniture.lua]: i think it's th next line: World.current.GetTileAt(jobSpot.X, jobSpot.y (! needs to be Y) -1)

Comment (sboigelot) [inline Assets/StreamingAssets/LUA/Furniture.lua]: > @sboigelot I suggest avoiding in progress PRs. People will test out your code when you're not finished yet, losing time.

I tend to agree, but here the help was really great. I would have lot much more time without all the other coders.

**Comment by sboigelot** [comment]
The feature is ready!

File: Assets/StreamingAssets/LUA/Furniture.lua [hunk 2]
```diff
@@ -307,4 +307,70 @@ function CloningPod_JobComplete(j)
...
+				inputSpot.inventory = nil
+			end
+            
+		end
+	end
+end
+
+function LandingPad_Temp_JobComplete(j)
+	jobSpot = j.furniture.GetJobSpotTile()
+	inputSpot = World.current.GetTileAt(jobSpot.X, jobSpot.y-1)
```
Comment (relinu) [inline Assets/StreamingAssets/LUA/Furniture.lua]: Does this work now? yous till have jobSpot.y instead of jobSpot.Y

**Comment by bjubes** [comment]
Is that s in comic sans?😂

**Comment by sboigelot** [comment]
@bjubes indeed :D you have the eye!

**Comment by vogonistic** [comment]
Awesome!

**Pull request merged**